x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agents/agent_list_page/components/migrate_agent_flyout/index.tsx

```diff
@@ -497,26 +565,39 @@ export const AgentMigrateFlyout: React.FC<Props> = ({ agents, onClose, onSave })
                             }
                           />
                         </EuiFlexItem>
-                        <EuiFlexItem>
-                          <EuiSwitch
-                            label={
-                              <FormattedMessage
-                                id="xpack.fleet.agentList.migrateAgentFlyout.replaceTokenLabel"
-                                defaultMessage="Replace Token"
-                              />
-                            }
-                            checked={formContent.settings?.replace_token ?? false}
-                            onChange={(e) =>
-                              setFormContent({
-                                ...formContent,
-                                settings: {
-                                  ...formContent.settings,
-                                  replace_token: e.target.checked,
-                                },
-                              })
-                            }
-                          />
-                        </EuiFlexItem>
+                        {/* Replace token shouldnt be an option when bulk migrating */}
+                        {agents.length === 1 && (
+                          <EuiFlexItem>
+                            <EuiSwitch
+                              data-test-subj="migrateAgentFlyoutReplaceTokenButton"
+                              label={
+                                <FormattedMessage
+                                  id="xpack.fleet.agentList.migrateAgentFlyout.replaceTokenLabel"
+                                  defaultMessage="Replace Token"
+                                />
+                              }
+                              checked={
+                                agents.length === 1
```

Comment:
the condition `agents.length === 1` seems unnecessary as it's already checked in line 569